### PR TITLE
Refactor backend code structure

### PR DESCRIPTION
**Pull request by gracez72**
Organized backend, no changes from dev branch functionality

### CODE REVIEW

File: private-templates-service/src/config/passport.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import config from './config';
+import logger from '../util/logger';
+
+enum LoggingLevels {
+    Info = 'info',
+    Warn = 'warn',
+    Error = 'error'
+}
+
+var options = {
```
Comment (Jackbk) [inline private-templates-service/src/config/passport.ts]: Is this a type given to us by passport?

File: private-templates-service/src/config/passport.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    isB2C: false,
+    policyName: undefined,
+    allowMultiAudiencesInToken: config.allowMultiAudiencesInToken,
+    audience: config.clientID,
+    loggingLevel: LoggingLevels.Info
+};
+
+var OIDCBearerStrategy = passportAD.BearerStrategy;
+var bearerStrategy = new OIDCBearerStrategy(options,
+    function(token: any, done: Function) {
```
Comment (Jackbk) [inline private-templates-service/src/config/passport.ts]: Token should have a type, and the done function should also have a type that enforces an interface

File: private-templates-service/src/util/logger.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import bunyan from 'bunyan';
```
Comment (Jackbk) [inline private-templates-service/src/util/logger.ts]: I'm glad you added a logging library :)

File: private-templates-service/src/config/passport.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    issuer: undefined,                                    // For tenant-specific endpoint, issuer from metadata is used by default
+    passReqToCallback: false,
+    isB2C: false,
+    policyName: undefined,
+    allowMultiAudiencesInToken: config.allowMultiAudiencesInToken,
+    audience: config.clientID,
+    loggingLevel: LoggingLevels.Info
+};
+
+var OIDCBearerStrategy = passportAD.BearerStrategy;
```
Comment (Jackbk) [inline private-templates-service/src/config/passport.ts]: Is there a reason for defining this seperately or can you make the next line passportAD.BearerStrategy(.......)?

File: private-templates-service/src/config/passport.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import passport from 'passport';
+import passportAD from 'passport-azure-ad';
+import config from './config';
+import logger from '../util/logger';
+
+enum LoggingLevels {
```
Comment (Jackbk) [inline private-templates-service/src/config/passport.ts]: Should probably be in Logger file

File: private-templates-service/src/config/passport.ts (hunk 1 above)
Comment (Jackbk) [inline private-templates-service/src/config/passport.ts]: Is this being used anywhere?

File: private-templates-service/src/config/passport.ts (hunk 5 above)
Comment (gracez72) [reply private-templates-service/src/config/passport.ts]: These are the levels specified by passport library, I added a comment to file.

File: private-templates-service/src/config/passport.ts (hunk 1 above)
Comment (Jackbk) [reply private-templates-service/src/config/passport.ts]: Found it

File: private-templates-service/src/config/passport.ts (hunk 4 above)
Comment (gracez72) [reply private-templates-service/src/config/passport.ts]: Thanks for the catch, fixed!

File: private-templates-service/src/config/passport.ts (hunk 2 above)
Comment (gracez72) [reply private-templates-service/src/config/passport.ts]: Fixed, thanks!

File: private-templates-service/src/config/passport.ts (hunk 1 above)
Comment (gracez72) [reply private-templates-service/src/config/passport.ts]: Yep

Comment (Jackbk) [reply private-templates-service/src/config/passport.ts]: Could you explicitly type it?

**Pull request merged**